### PR TITLE
Test with Jest

### PR DESCRIPTION
**Pull request by billodiallo**
step 1
install jest in your project and create test folder 
- Add files that need to be tested in test folder.
- write the code to test the code for each feature.
- npm i jest --save-dev
npm install --save-dev @babel/preset-env
- test todo.js
- test project.js
- test utility.js
- add screenshot

### CODE REVIEW

File: test/todo.spec.js [hunk 1]
```diff
@@ -18,21 +18,21 @@ describe('change attr', () => {
...
 });
 
 describe('format Props', () => {
   test('it returns an array with the instance values formatted', () => {
-    expect(myTodo.formatProps()).toEqual(['myTodoTitle', 'myDescription', '2021-05-11', 'Medium', 'myTitle']);
+    expect(myTodo.formatProps()).toEqual(['myTodoTitle', 'myDescription', '2021-05-17', 'Medium', 'myTitle']);
   });
-});
\ No newline at end of file
+});
```
Comment (bellom) [inline test/todo.spec.js]: We need to write test for the following methods;
1. `addToProject`
2. `editTask`

File: test/utility.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    classList: {
+      toggle: jest.fn(),
+    },
+  };
+
+  it('should call toggle with "hide"', () => {
+    utility.toggleShowElement(mockElement);
+    expect(mockElement.classList.toggle).toBeCalledWith('hide');
+  });
+});
```
Comment (bellom) [inline test/utility.spec.js]: You need to write more test in this file, you only wrote a test for just one method which is `toggleShowElement `, kindly write test for the remaining methods in `utility.js`.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @billodiallo  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

Heads-Up: Always double-check your commit messages, you have unnecessary white space in one of the commit messages.

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by billodiallo** [comment]
@bellom 
I tried to test those method , it's not working as expecting , I spent many hours on them . I think i completed most of the requirements .
if we can set up a zoom meeting to write those test together or TSEs .
I look forward to hearing from you.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hello Team :wave:
Status: Changes Required :recycle:
To Highlight: :angel:

- Modularization: Separate file is created for each class and module :+1: 

Great job so far! :clap: :wave:

- [ ] [REQUIRED] Please work on previous TSE requirements. In fact, All functions should be tested.

:camera_flash: Screenshot from your project, test results with coverage.


- You can use the --coverage flag to get information

That's it, you can submit another code review request after making these changes :recycle: :innocent:

With all my respect :pray:

**Comment by bellom** [comment]
> @bellom
> I tried to test those method , it's not working as expecting , I spent many hours on them . I think i completed most of the requirements .
> if we can set up a zoom meeting to write those test together or TSEs .
> I look forward to hearing from you.

I replied to your message on slack, kindly check.

**Comment by billodiallo** [comment]
Before you review this project  kindly contact @Christian Otieno on slack or TSEs channel to have a better understanding and failing and remaining function not tested.
Because of encapsulation of some methods, some test instances won't work they are not a function that can test expect for some outputs. We spent 1 hour together exploring all the possibilities, so the testable functions are working as I said in the previous.
Thanks in advance to Microverse TSEs for this innovation, getting the opportunity to talk with a TSE, explain to them the challenge you are facing and solve them with different scenarios. It helps Microverse to save time on certain issues which are out of our control. Thanks, looking forward to getting approved so I can move to the next step.

**Comment by bellom** [comment]
@current TSE, kindly approve I and @christian Otieno already look into the issue.

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**

- [ ] OPTIONAL - This should be required but since it wasn't mentioned in previous reviews it's been marked as optional. Kindly ensure you use the same eslint config rules as shown in this [link](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json).

Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**